### DISCUSSION TITLE
feature request : GUI  for including navigation for whatever website interests the user

### DISCUSSION THREAD

**Discussion by userrand** [infokiller/web-search-navigator#516] [Ideas]
_The highlighted text in bold is sufficient to read._  

If possible it would be nice to include a GUI that allows the user to implement cursor navigation for whatever page they would like. 

Basically, **the user chooses 1 to 3 examples of the element they would like to scroll through by clicking on them and the add-on finds the common patterns** (class or id in the HTML for example) and then the addon implements easy navigation on that site.

There might be an advantage both for the user and for the developers as this avoids users having to ask feature requests for every website that interests them or having to ask whenever a website changes significantly. 

I am currently learning Javascript, I am a beginner in Python, rusty in Fortran, and advanced in Mathematica. I can maybe implement this myself as a fun project with the information provided in the development section of the readme. However, if a developer wishes to implement this then please feel free to as it might take me some time (perhaps months) given my experience, knowledge, and constraints.

**Comment by infokiller** [comment]
Cool idea. I won't get to this anytime soon, but feel free to send PRs (if you do major changes to the code, please discuss before)

**Comment by userrand** [comment]
Update: no update (: . Just mentioning that I have not really started trying to code this yet. I am probably going to take a while before I even start. If anyone reading this wants to try coding this feel free to do so.
